products/views.py

```diff
@@ -1,3 +1,49 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+from django.db    import transaction, IntegrityError
+
+from products.models import *
+from products.utils  import login_decorator
+
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        try:
+            product = Product.objects.get(id = product_id)
+
+            product_color_images = product.product_products_colors_images.all()
+
+            colors = [{
+                'id'        : product_color.color.id,
+                'name'      : product_color.color.name,
+                'image_url' : product_color.image_url
+            } for product_color in product_color_images ]
+
+            product_detail ={
+                'product_id' : product.id,
+                'name'       : product.name,
+                'price'      : product.price,
+                'colors'     : colors
+            }
+
+            return JsonResponse({"results": product_detail}, status=201)
+
+        except Product.DoesNotExist:
+            return JsonResponse({"message": "DOES_NOT_EXIST"}, status=400)
+
+class SizeView(View):
+    def get(self, request):
+        try:
+
+            sizes_db = Size.objects.all()
+            
+            sizes =[{ 
+                'id'   : size_db.id,
+                'size' : size_db.size
+            } for size_db in sizes_db ]
+
+            return JsonResponse({"results": sizes}, status=201)
+        
+        except Size.DoesNotExist:
+            return JsonResponse({"message": "DOES_NOT_EXIST"}, status=400)
```

Comment:
위 코드를 한 줄로 줄여볼 수 있을 것 같습니다.
추가로, 위 코드에서는 Size.DoesNotExist 에러가 발생할 수 있는 코드가 없는데 어떤 의도로 추가하셨나요?

```suggestion
    def get(self, request):
        return JsonResponse({"sizes" : Size.objects.values("id", "size")}, status=200)
```

Reply:
앗 위의 코드를 가져와서 수정해보다보니 작성된것같습니다!
리뷰해주신 수정사항에 맞게 수정해서 다시 pr드리겠습니다.